staging/src/k8s.io/apiserver/pkg/storage/etcd3/latency_tracker.go

```diff
@@ -0,0 +1,91 @@
+package etcd3
+
+import (
+	"context"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	endpointsrequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// NewETCDLatencyTracker returns an implementation of
+// clientv3.KV that times the calls from the specified
+// 'delegate' KV instance in order to track latency incurred.
+func NewETCDLatencyTracker(delegate clientv3.KV) clientv3.KV {
```

Comment:
instead of tapping into `storage.Interface`, I am now tapping into etcd3 client `KV` so we can measure latency from etcd3 round trips only